Update Genplugin to support 1st and 3rd party plugins

Coffee2Bits opened the pull request:
Closes #142.

* Genplugin now has a second argument for first or third party plugin
* Fixed a template output bug; it will now generate **_multiple_** levels of directories
* Added some help command formatting so that it's less cramped/stacked
* Added a `choices` to the AaCCommand class so that plugins can provide a set of allowable choices
* Updated functional test workflow to test 1st and 3rd party plugin generation 
* Update GenPlugin model with new behavior
* Removed the check to have all AaC plugins prefixed with "aac"
* Updated the templates to be aligned with current implementation patterns.

Coffee2Bits commented on .vscode/python.code-workspace:
```diff
@@ -18,7 +18,10 @@
-        "python.pythonPath": "python/venv/bin/python"
+        "python.pythonPath": "python/venv/bin/python",
+        "python.formatting.blackArgs": [
+            "--line-length=127"
```
I'm using the same value here as what's specified in tox.ini, and we should keep them synced.

Personally, I'm against any strict enforcement of line length; I think it should be up to the developer to maintain what's most readable just like they do for whitespace and the general construction of the code. However, having spent some time in reality, I agree with the obvious benefits of having a defined standard with enforcement for a consistent style like what we have here. Reducing the maximum line length to something like 99 is too aggressive in my opinion. I personally find it difficult to read aggressively shortened code, and I don't think that it adds enough value to accommodate some tools' gap in modern accessibility features like zoom or custom font size.

Coffee2Bits commented on python/src/aac/templates/genplug/third_party/plugin_impl.py.jinja2:
```diff
@@ -0,0 +1,39 @@
+"""AaC Plugin implementation module for the {{plugin.name}} plugin."""
+# NOTE: It is safe to edit this file.
+# This file is only initially generated by the aac gen-plugin, and it won't be overwritten if the file already exists.
```
I think it sounds better with the "the" present.

cvchaparro replied:
If we leave the "the" in, I think it makes more sense to say "... generated by the aac gen-plugin plugin ...".

Coffee2Bits commented:
I liked Cameron's advice, so I changed the plugin a bit to automatically detect and generate first-party plugins.

Additional changes:
* Cleaned up the Gen-plugin plugin
* Removed the check to have all AaC plugins prefixed with "aac"
* Updated the templates to be aligned with current implementation patterns.

Coffee2Bits commented on python/src/aac/plugins/gen_plugin/__init__.py:
```diff
@@ -21,7 +21,7 @@ def get_commands() -> list[AacCommand]:
             "architecture_file",
             "The yaml file containing the AaC DSL of the plugin architecture.",
-        )
+        ),
```
```suggestion
        )
```

cvchaparro commented on python/src/aac/templates/genplug/first_party/__init__.py.jinja2:
```diff
@@ -0,0 +1,60 @@
+
+
+@hookimpl
+def get_base_model_extensions() -> str:
```
```suggestion
def get_plugin_aac_definitions() -> str:
```

cvchaparro commented on python/src/aac/templates/genplug/first_party/plugin_impl.py.jinja2:
```diff
@@ -31,9 +31,9 @@ def {{command.implementation_name}}({% for command_input in command.input -%}{{c
     """
     # TODO add implementation here
-    def implement_me():
+    def _implement__and_rename_me():
```
```suggestion
    def _implement_and_rename_me():
```

cvchaparro reviewed (approved):
Overall looks good. Nice work.
